File: ncPersistencia/pUsuario.vb
Imports ncDados.nsUsuario
Imports ncComum.nsAcessoBD
Imports ncComum.nsFuncoes
Imports ncComum.nsExcecao

Namespace nsUsuario

  Public Class pUsuario

    Public Function Listar() As colecaoUsuario

      Dim retorno As colecaoUsuario
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dUsuario
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD


                comandoSQL = " Select cid, nomeCompleto, senha, situacao, usuario, descontoProduto, descontoPedido, " &
                    " comissao, usuarioPerfil_cid, email From usuarios where situacao like 'A'"

                ds = acessoBanco.ExecutarDS(comandoSQL)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoUsuario

              For Each row In dt.Rows
                item = New dUsuario

                                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                                item.nomeCompleto = cFuncoes.RetornarTexto(row("nomeCompleto"))
                                item.senha = cFuncoes.RetornarTexto(row("senha"))
                                item.situacao = cFuncoes.RetornarTexto(row("situacao"))
                                item.usuario = cFuncoes.RetornarTexto(row("usuario"))
                                item.descontoProduto = cFuncoes.RetornarDecimal(row("descontoProduto"))
                                item.descontoPedido = cFuncoes.RetornarDecimal(row("descontoPedido"))
                                item.comissao = cFuncoes.RetornarDecimal(row("comissao"))
                                item.usuarioPerfil_cid = cFuncoes.RetornarInteiro(row("usuarioPerfil_cid"))
                                item.Email = cFuncoes.RetornarTexto(row("email"))
                                retorno.Add(item)
                            Next
                        Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Listar Usuario [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Listar = retorno

    End Function

    Public Function Consultar(ByVal dados As dUsuario) As ColecaoUsuario

      Dim retorno As ColecaoUsuario
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dUsuario
      Dim sqlSelect As String
      Dim sqlWhere As String
      Dim sqlFrom As String

      Try

        acessoBanco = New cAcessoBD

                sqlSelect = " Select u.cid, u.nomeCompleto, u.senha, u.situacao, u.usuario, u.usuarioPerfil_cid, " &
            " up.codigo, u.descontoPedido, u.descontoProduto, u.comissao, email "
                sqlWhere = String.Empty
        sqlFrom = " From usuarios u INNER JOIN usuarioperfil up " & _
            " ON up.cid = u.usuarioPerfil_cid "

        '-- cid
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.cid, "u.cid")

        '-- usuario
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.usuario, "u.usuario")

        '-- senha
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.senha, "u.senha")

        '-- situacao
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.situacao, "u.situacao")

        '-- nomeCompleto
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.nomeCompleto, "u.nomeCompleto")

        '-- usuarioPerfil_cid
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.usuarioPerfil_cid, "u.usuarioPerfil_cid")

        '-- descontoProduto
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.descontoProduto, "u.descontoProduto")

        '-- descontoPedido
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.descontoPedido, "u.descontoPedido")

                '-- comissao
                sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.comissao, "u.comissao")

                '-- email
                sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.Email, "u.email")

                If Not sqlWhere.Equals(String.Empty) Then
          sqlWhere = " WHERE " & sqlWhere
        End If

        ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom & " " & sqlWhere)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoUsuario

              For Each row In dt.Rows
                                item = New dUsuario
                                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                                item.nomeCompleto = cFuncoes.RetornarTexto(row("nomeCompleto"))
                                item.senha = cFuncoes.RetornarTexto(row("senha"))
                                item.situacao = cFuncoes.RetornarTexto(row("situacao"))
                                item.usuario = cFuncoes.RetornarTexto(row("usuario"))
                                item.usuarioPerfil_cid = cFuncoes.RetornarInteiro(row("usuarioPerfil_cid"))
                                item.usuarioPerfil_codigo = cFuncoes.RetornarTexto(row("codigo"))
                                item.descontoProduto = cFuncoes.RetornarDecimal(row("descontoProduto"))
                                item.descontoPedido = cFuncoes.RetornarDecimal(row("descontoPedido"))
                                item.comissao = cFuncoes.RetornarDecimal(row("comissao"))
                                item.Email = cFuncoes.RetornarTexto(row("email"))
                                retorno.Add(item)
                            Next
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception
        retorno = Nothing
        Throw ex
        Throw New ExcecaoNascomercio("Erro em Consultar Usuario [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Consultar = retorno

    End Function
        Public Function ConsultarADM(ByVal perfil As String) As ColecaoUsuario

            Dim retorno As ColecaoUsuario
            Dim acessoBanco As cAcessoBD
            Dim ds As DataSet
            Dim dt As DataTable
            Dim row As DataRow
            Dim item As dUsuario
            Dim sqlSelect As String
            Dim sqlWhere As String
            Dim sqlFrom As String


            Try


                acessoBanco = New cAcessoBD

                sqlSelect = " Select u.cid, u.nomeCompleto, u.senha, u.situacao, u.usuario, u.usuarioPerfil_cid, " &
                " up.codigo, u.descontoPedido, u.descontoProduto, u.comissao, u.email "
                sqlWhere = String.Empty
                sqlFrom = " From usuarios u INNER JOIN usuarioperfil up " &
                " ON up.cid = u.usuarioPerfil_cid " &
                $"where up.nome = '{perfil}'" &
                "and u.email IS NOT NULL;"

                If Not sqlWhere.Equals(String.Empty) Then
                    sqlWhere = " WHERE " & sqlWhere
                End If

                ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom & " " & sqlWhere)

                If Not ds Is Nothing Then
                    If ds.Tables.Count > 0 Then
                        dt = ds.Tables(0)

                        If dt.Rows.Count > 0 Then
                            retorno = New ColecaoUsuario

                            For Each row In dt.Rows
                                item = New dUsuario
                                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                                item.nomeCompleto = cFuncoes.RetornarTexto(row("nomeCompleto"))
                                item.senha = cFuncoes.RetornarTexto(row("senha"))
                                item.situacao = cFuncoes.RetornarTexto(row("situacao"))
                                item.usuario = cFuncoes.RetornarTexto(row("usuario"))
                                item.usuarioPerfil_cid = cFuncoes.RetornarInteiro(row("usuarioPerfil_cid"))
                                item.usuarioPerfil_codigo = cFuncoes.RetornarTexto(row("codigo"))
                                item.descontoProduto = cFuncoes.RetornarDecimal(row("descontoProduto"))
                                item.descontoPedido = cFuncoes.RetornarDecimal(row("descontoPedido"))
                                item.comissao = cFuncoes.RetornarDecimal(row("comissao"))
                                item.Email = cFuncoes.RetornarTexto(row("email"))
                                retorno.Add(item)
                            Next
                        Else
                            retorno = Nothing
                        End If
                    Else
                        retorno = Nothing
                    End If
                Else
                    retorno = Nothing
                End If

            Catch ex As Exception
                retorno = Nothing
                Throw ex
                Throw New ExcecaoNascomercio("Erro em Consultar Usuario [" & Me.ToString() & "] - " & ex.Message)

            End Try

            ConsultarADM = retorno

        End Function
        Public Function Incluir(ByVal dados As dUsuario) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String


      Try

        acessoBanco = New cAcessoBD


                comandoSQL = " INSERT INTO " &
            " usuarios (usuario, senha, nomeCompleto, situacao, descontoProduto, descontoPedido, " &
            " comissao, email, usuarioPerfil_cid) " &
            " VALUES (" &
            cFuncoes.PersistirTexto(dados.usuario) & "," &
            cFuncoes.PersistirTexto(dados.senha) & "," &
            cFuncoes.PersistirTexto(dados.nomeCompleto) & "," &
            cFuncoes.PersistirTexto(dados.situacao) & "," &
            cFuncoes.PersistirDecimal(dados.descontoProduto) & "," &
            cFuncoes.PersistirDecimal(dados.descontoPedido) & "," &
            cFuncoes.PersistirDecimal(dados.comissao) & "," &
            cFuncoes.PersistirTexto(dados.Email) & "," &
            cFuncoes.PersistirInteiro(dados.usuarioPerfil_cid) & ")"

                retorno = acessoBanco.ExecutarCID(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Incluir Usuario [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Incluir = retorno

    End Function

    Public Function Alterar(ByVal dados As dUsuario) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String


      Try

        acessoBanco = New cAcessoBD


                comandoSQL = " UPDATE usuarios SET " &
            " usuario = " & cFuncoes.PersistirTexto(dados.usuario) & "," &
            " senha = " & cFuncoes.PersistirTexto(dados.senha) & "," &
            " nomeCompleto = " & cFuncoes.PersistirTexto(dados.nomeCompleto) & "," &
            " situacao = " & cFuncoes.PersistirTexto(dados.situacao) & "," &
            " descontoProduto = " & cFuncoes.PersistirDecimal(dados.descontoProduto) & "," &
            " descontoPedido = " & cFuncoes.PersistirDecimal(dados.descontoPedido) & "," &
            " comissao = " & cFuncoes.PersistirDecimal(dados.comissao) & "," &
            " usuarioPerfil_cid = " & cFuncoes.PersistirInteiro(dados.usuarioPerfil_cid) & "," &
            " email = " & cFuncoes.PersistirTexto(dados.Email) &
            " WHERE " &
            " cid = " & dados.cid.ToString()

                retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Alterar Usuario [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Alterar = retorno

    End Function

    Public Function Excluir(ByVal dados As dUsuario) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " DELETE FROM usuarios " & _
            " WHERE cid = " & dados.cid.ToString()

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Excluir Usuario [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Excluir = retorno

    End Function

  End Class

End Namespace

File: ncRepositorios/pUsuario.cs
using System;
using System.Collections.Generic;
using Dapper;
using MySql.Data.MySqlClient;
using ncNComum.nsExcecao;
using nsModelos;

namespace ncPersistencia.nsUsuario
{
    public class pUsuario : RepositorioBase, IpUsuario
    {
        public ColecaoUsuario Listar()
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var lista = conn.Query<dUsuario>(
                        "SELECT cid, nomeCompleto, senha, situacao, usuario, descontoProduto, descontoPedido, comissao, usuarioPerfil_cid, email AS Email FROM usuarios WHERE situacao LIKE 'A'").AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoUsuario();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Listar Usuario [" + ToString() + "] - " + ex.Message);
            }
        }

        public ColecaoUsuario Consultar(dUsuario dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var conditions = new List<string>();
                    var p = new DynamicParameters();
                    if (dados.cid != 0) { conditions.Add("u.cid=@cid"); p.Add("cid", dados.cid); }
                    if (!string.IsNullOrEmpty(dados.usuario)) { conditions.Add("u.usuario=@usuario"); p.Add("usuario", dados.usuario); }
                    if (!string.IsNullOrEmpty(dados.senha)) { conditions.Add("u.senha=@senha"); p.Add("senha", dados.senha); }
                    if (!string.IsNullOrEmpty(dados.situacao)) { conditions.Add("u.situacao=@situacao"); p.Add("situacao", dados.situacao); }
                    if (!string.IsNullOrEmpty(dados.nomeCompleto)) { conditions.Add("u.nomeCompleto=@nomeCompleto"); p.Add("nomeCompleto", dados.nomeCompleto); }
                    if (dados.usuarioPerfil_cid != 0) { conditions.Add("u.usuarioPerfil_cid=@usuarioPerfil_cid"); p.Add("usuarioPerfil_cid", dados.usuarioPerfil_cid); }
                    if (dados.descontoProduto != 0) { conditions.Add("u.descontoProduto=@descontoProduto"); p.Add("descontoProduto", dados.descontoProduto); }
                    if (dados.descontoPedido != 0) { conditions.Add("u.descontoPedido=@descontoPedido"); p.Add("descontoPedido", dados.descontoPedido); }
                    if (dados.comissao != 0) { conditions.Add("u.comissao=@comissao"); p.Add("comissao", dados.comissao); }
                    if (!string.IsNullOrEmpty(dados.Email)) { conditions.Add("u.email=@email"); p.Add("email", dados.Email); }
                    var where = conditions.Count > 0 ? "WHERE " + string.Join(" AND ", conditions) : "";
                    var sql = $"SELECT u.cid, u.nomeCompleto, u.senha, u.situacao, u.usuario, u.usuarioPerfil_cid, up.codigo AS usuarioPerfil_codigo, u.descontoPedido, u.descontoProduto, u.comissao, u.email AS Email FROM usuarios u INNER JOIN usuarioperfil up ON up.cid = u.usuarioPerfil_cid {where}";
                    var lista = conn.Query<dUsuario>(sql, p).AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoUsuario();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public ColecaoUsuario ConsultarADM(string perfil)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var p = new DynamicParameters();
                    p.Add("perfil", perfil);
                    var sql = "SELECT u.cid, u.nomeCompleto, u.senha, u.situacao, u.usuario, u.usuarioPerfil_cid, up.codigo AS usuarioPerfil_codigo, u.descontoPedido, u.descontoProduto, u.comissao, u.email AS Email FROM usuarios u INNER JOIN usuarioperfil up ON up.cid = u.usuarioPerfil_cid WHERE up.nome=@perfil AND u.email IS NOT NULL";
                    var lista = conn.Query<dUsuario>(sql, p).AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoUsuario();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public int Incluir(dUsuario dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    conn.Execute("INSERT INTO usuarios (usuario, senha, nomeCompleto, situacao, descontoProduto, descontoPedido, comissao, email, usuarioPerfil_cid) VALUES (@usuario, @senha, @nomeCompleto, @situacao, @descontoProduto, @descontoPedido, @comissao, @Email, @usuarioPerfil_cid)",
                        new { dados.usuario, dados.senha, dados.nomeCompleto, dados.situacao, dados.descontoProduto, dados.descontoPedido, dados.comissao, dados.Email, dados.usuarioPerfil_cid });
                    return (int)conn.ExecuteScalar<long>("SELECT LAST_INSERT_ID()");
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Incluir Usuario [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Alterar(dUsuario dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute("UPDATE usuarios SET usuario=@usuario, senha=@senha, nomeCompleto=@nomeCompleto, situacao=@situacao, descontoProduto=@descontoProduto, descontoPedido=@descontoPedido, comissao=@comissao, usuarioPerfil_cid=@usuarioPerfil_cid, email=@Email WHERE cid=@cid",
                        new { dados.usuario, dados.senha, dados.nomeCompleto, dados.situacao, dados.descontoProduto, dados.descontoPedido, dados.comissao, dados.usuarioPerfil_cid, dados.Email, dados.cid });
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Alterar Usuario [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Excluir(dUsuario dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute("DELETE FROM usuarios WHERE cid=@cid", new { dados.cid });
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Excluir Usuario [" + ToString() + "] - " + ex.Message);
            }
        }
    }
}
